GetStream/stream-chat-android
# [CAS-881] Attachment db updates

### 🎯 Goal

Created `MessageSendingService` that send message and observe DB for attachments sending result.

### 🛠 Implementation details

1. Created `MessageSendingService` as an encapsulation of sending logic for messages.
2. Deprecated `sendMessage` with `attachmentTransformer`
3. Removed new field in `Message::attachmentsSyncStatus` (we just released it, it's redundant. I hope no one used it)
4. Added new `WAIT_ATTACHMENTS` SyncStatus value
5. `callAndRetry` moved from `ChatDomain` to utility `CallRetryService`
6. Unit tests :-)

### 🎨 UI Changes

https://user-images.githubusercontent.com/11807573/121910382-18503a00-cd2f-11eb-97ee-21af0690ff94.mov

### 🧪 Testing

1) Send attachment while being offline
2) Switch to online mode
3) See that message and attachments are sent

### ☑️ Checklist


### 🎉 GIF

![](https://media4.giphy.com/media/hWclJYuJA7aJMT8abO/giphy.gif)
